functions/bigtable/main_async.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START bigtable_functions_quickstart_asyncio]
+import asyncio
+
+import functions_framework
+from google.cloud.bigtable.data import BigtableDataClientAsync, ReadRowsQuery, RowRange
+
+event_loop = asyncio.new_event_loop()
+asyncio.set_event_loop(event_loop)
+
+
+# Setup: create a shared client within the context of the event loop
+async def create_client():
+    # create client in the asyncio event loop context to
+    # give background tasks a chance to initialize
+    return BigtableDataClientAsync()
+
+
+client = event_loop.run_until_complete(create_client())
+
+
+# Actual cloud functions entrypoint, will delegate to the async one
+@functions_framework.http
+def bigtable_read_data(request):
+    return event_loop.run_until_complete(_bigtable_read_data_async(request))
```

Comment:
Same here, would it be possible to use await somehow?

Reply:
If `functions_framework.http` requires a non-async function it might not be possible, but can we at least add a comment mentioning that?

Reply:
Yeah, the problem is that functions_framework can only execute sync functions, so we have to use the sync function as launcher for the async event loop